tests/device.test.js

```diff
@@ -146,6 +146,140 @@ describe(OlmMachine.name, () => {
+                                    "ed25519:AFGUOBTZWM":
+                                        "RoSWvru1jj6fs2arnTedWsyIyBmKHMdOu7r9gDi0BZ61h9SbCK2zLXzuJ9ZFLao2VvA0yEd7CASCmDHDLYpXCA",
+                                },
+                            },
+                            user_id: "@example:localhost",
+                            unsigned: {
+                                device_display_name: "rust-sdk",
+                            },
+                        },
+                    },
+                },
+                failures: {},
+                master_keys: {
+                    "@example:localhost": {
+                        user_id: "@example:localhost",
+                        usage: ["master"],
+                        keys: {
+                            "ed25519:n2lpJGx0LiKnuNE1IucZP3QExrD4SeRP0veBHPe3XUU":
+                                "n2lpJGx0LiKnuNE1IucZP3QExrD4SeRP0veBHPe3XUU",
+                        },
+                        signatures: {
+                            "@example:localhost": {
+                                "ed25519:TCSJXPWGVS":
+                                    "+j9G3L41I1fe0++wwusTTQvbboYW0yDtRWUEujhwZz4MAltjLSfJvY0hxhnz+wHHmuEXvQDen39XOpr1p29sAg",
+                            },
+                        },
+                    },
+                },
+                self_signing_keys: {
+                    "@example:localhost": {
+                        user_id: "@example:localhost",
+                        usage: ["self_signing"],
+                        keys: {
+                            "ed25519:kQXOuy639Yt47mvNTdrIluoC6DMvfbZLYbxAmwiDyhI":
+                                "kQXOuy639Yt47mvNTdrIluoC6DMvfbZLYbxAmwiDyhI",
+                        },
+                        signatures: {
+                            "@example:localhost": {
+                                "ed25519:n2lpJGx0LiKnuNE1IucZP3QExrD4SeRP0veBHPe3XUU":
+                                    "q32ifix/qyRpvmegw2BEJklwoBCAJldDNkcX+fp+lBA4Rpyqtycxge6BA4hcJdxYsy3oV0IHRuugS8rJMMFyAA",
+                            },
+                        },
+                    },
+                },
+                user_signing_keys: {
+                    "@example:localhost": {
+                        user_id: "@example:localhost",
+                        usage: ["user_signing"],
+                        keys: {
+                            "ed25519:g4ED07Fnqf3GzVWNN1pZ0IFrPQVdqQf+PYoJNH4eE0s":
+                                "g4ED07Fnqf3GzVWNN1pZ0IFrPQVdqQf+PYoJNH4eE0s",
+                        },
+                        signatures: {
+                            "@example:localhost": {
+                                "ed25519:n2lpJGx0LiKnuNE1IucZP3QExrD4SeRP0veBHPe3XUU":
+                                    "nKQu8alQKDefNbZz9luYPcNj+Z+ouQSot4fU/A23ELl1xrI06QVBku/SmDx0sIW1ytso0Cqwy1a+3PzCa1XABg",
+                            },
+                        },
+                    },
+                },
```

Comment:
for simplicity, could we omit the x-signing keys (and the x-signing signature above) from this response?

Reply:
yes good catch